### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by TomaszEichler**

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -4,5 +4,12 @@
 int addEven(const std::vector<int>& numbers) {
     // TODO: Your implementation goes here
     // Below return is only to make this function compile now
-    return -1;
+    int sum {};
+
+    for(const auto& element : numbers){
+        if(!(element % 2)){
```
Comment (piezol) [inline module2/homework/addEven/addEven.hpp:10]: Implicit casts are usually a code smell.
Comparison to 0 is much clearer
```suggestion
        if(element % 2 == 0) {
```

File: module2/homework/generateSequence/generateSequence.hpp [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    std::vector<int> myVector{};
+
+    if(count <= 0){
+        return myVector;
+    }
+
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:10]: Try to preallocate memory for the whole vector upfront - you know the size of the vector at the end.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+        return rhs;
+    }
+    else if(rhs == 0){
+        return lhs;
+    }
+
+    int temp;
+    while(rhs != 0){
+        temp = std::abs(rhs);
+        rhs = std::abs(lhs) % temp;
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:15]: You have two calls two abs in every iteration.
The algorithm will work correctly if you call abs before the loop and always work on positive values.

Comment (TomaszEichler) [reply module2/homework/nwdNww/nwdNww.hpp:15]: Thank you, great idea :)

File: module2/homework/generateSequence/generateSequence.hpp (hunk 2 above)
Comment (TomaszEichler) [reply module2/homework/generateSequence/generateSequence.hpp:10]: Yeah, you're right I should do that after `if` condition. Good advice, thank you :)

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05

{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**